source/agora/node/Validator.d

```diff
@@ -411,4 +415,29 @@ public class Validator : FullNode, API
         super.shutdown();
         this.nominator.storeLatestState();
     }
+
+    /***************************************************************************
+
+        Check if the current enrollment is about to expire and re-enroll
+        if necessary.
+
+        Params:
+            block_height = Current block height
+
+    ***************************************************************************/
+
+    private void checkEnrollmentExpiration (Height block_height) @safe
+    {
+        Hash enroll_key = this.enroll_man.getEnrolledUTXO(this.utxo_set.getUTXOFinder());
+        if (enroll_key != Hash.init)
```

Comment:
Personal preference by me & Mathias is to prefer early returns rather than nesting indentation.

So:

```D
if (enroll_key == Hash.init)
    return;   // not enrolled
```